File: backend/src/main/java/com/simats/neoomfs/dto/request/LaboratoryRequest.java
package com.simats.neoomfs.dto.request;

import lombok.Data;

import java.math.BigDecimal;

/**
 * Inbound DTO for laboratory investigations (Step 4 of the wizard).
 */
@Data
public class LaboratoryRequest {
    private BigDecimal hemoglobin;
    private Integer totalWbcCount;
    private Integer plateletCount;
    private BigDecimal bleedingTime;
    private BigDecimal clottingTime;
    private BigDecimal pt;
    private BigDecimal inr;
    private BigDecimal aptt;
    private BigDecimal fastingBloodSugar;
    private BigDecimal randomBloodSugar;
    private BigDecimal hba1c;
    private BigDecimal bloodUrea;
    private BigDecimal serumCreatinine;
    private BigDecimal serumBilirubinTotal;
    private BigDecimal sgot;
    private BigDecimal sgpt;
    private String bloodGroup;
    private String rhFactor;
    private String hivStatus;
    private String hbsagStatus;
    private String hcvStatus;
    private String labReportFileUrl;
    private String notes;
}

File: frontend/app/src/main/java/com/simats/neoomfs/models/LaboratoryRequest.kt
package com.simats.neoomfs.models

data class LaboratoryRequest(
    val hemoglobin: Double?,
    val totalWbcCount: Int?,
    val plateletCount: Int?,
    val bleedingTime: Double?,
    val clottingTime: Double?,
    val pt: Double?,
    val inr: Double?,
    val aptt: Double? = null,
    val fastingBloodSugar: Double?,
    val randomBloodSugar: Double?,
    val hba1c: Double? = null,
    val bloodUrea: Double? = null,
    val serumCreatinine: Double? = null,
    val serumBilirubinTotal: Double? = null,
    val sgot: Double? = null,
    val sgpt: Double? = null,
    val bloodGroup: String?,
    val rhFactor: String? = null,
    val hivStatus: String? = null,
    val hbsagStatus: String? = null,
    val hcvStatus: String? = null,
    val labReportFileUrl: String? = null,
    val notes: String? = null
)
